tests/functional/test_accounts.py

```diff
@@ -147,6 +148,23 @@ def test_deploy_and_not_publish(mocker, owner, contract_container, dummy_live_ne
     assert not mock_explorer.call_count
 
 
+def test_deploy_proxy(
+    owner, project, vyper_contract_instance, proxy_contract_container, chain, eth_tester_provider
+):
+    target = vyper_contract_instance.address
+    proxy = owner.deploy(proxy_contract_container, target)
+    assert proxy.address in chain.contracts._local_contract_types
+    assert proxy.address in chain.contracts._local_proxies
+
+    actual = chain.contracts._local_proxies[proxy.address]
+    assert actual.target == target
+    assert actual.type == ProxyType.Delegate
+
+    # Show we get the implementation contract type using the proxy address
+    implementation = chain.contracts.instance_at(proxy.address)
```

Comment:
@johnson2427 I added this part of the test to further demonstrate the bugfix here.
So now, we pass the locally deployed proxy contract address to `chain.instance_at()` (same as `ape.Contract()`) and we are given back the implementation contract type.

Reply:
This is great work here. well done